### PR TITLE
feat: add feature flag recomposition support

### PR DESCRIPTION
**Pull request by wilsonrivera**
## Summary by CodeRabbit

* **New Features**
  * Added a recompose feature-flag capability: CLI command plus platform API to trigger recomposition/redeployment with namespace, limit, validation and warning controls.
* **Tests**
  * Added unit and integration tests covering success, error, warning and flag/namespace edge cases.
* **Chores**
  * Updated RPC surface, message types, client bindings, server handler wiring, and audit logging to support the new operation.

[Review Change Stack](https://app.coderabbit.ai/change-stack/wundergraph/cosmo/pull/2853)

## Checklist

- [ ] I have discussed my proposed changes in an issue and have received approval to proceed.
- [x] I have followed the coding standards of the project.
- [x] Tests or benchmarks have been added or updated.
- [ ] Documentation has been updated on [https://github.com/wundergraph/docs-website](https://github.com/wundergraph/docs-website).
- [x] I have read the [Contributors Guide](https://github.com/wundergraph/cosmo/blob/main/CONTRIBUTING.md).

## Open Source AI Manifesto

This project follows the principles of the [Open Source AI Manifesto](https://human-oss.dev). Please ensure your contribution aligns with its principles.

The goal of this PR is to introduce support to recompose specific feature flags the same way `federated-graph recompose` and `monograph recompose` works

### CODE REVIEW

File: cli/src/commands/feature-flag/commands/recompose.ts [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+      {
+        disableResolvabilityValidation: options.disableResolvabilityValidation,
+        limit,
+        name,
+        namespace: options.namespace,
+      },
+      {
+        headers: getBaseHeaders(),
+      },
+    );
```
[the remark this answers is not in the document]
Comment (wilsonrivera) [reply cli/src/commands/feature-flag/commands/recompose.ts:55]: We have a global error handler that catches any thrown error so we don't need to do it here

File: cli/src/commands/feature-flag/commands/recompose.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    '--fail-on-composition-error',
+    'If set, the command will fail if the composition of the feature flag fails.',
+    false,
+  );
+  command.option(
+    '--fail-on-admission-webhook-error',
+    'If set, the command will fail if the admission webhook fails.',
+    false,
+  );
+  command.option(
```
Comment (comatory) [inline cli/src/commands/feature-flag/commands/recompose.ts:30]: What about `--json`? The `handleCompositionResult` can already process output as JSON. Maybe we could at least add it as a follow up but it shouldn't take a lot of effort to add it in.

File: cli/src/commands/feature-flag/commands/recompose.ts [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        if (resp.response.details) {
+          message += `\n${pc.red(pc.bold(resp.response.details))}`;
+        }
+
+        program.error(message);
+        break;
+      }
+    }
+
+    try {
```
Comment (comatory) [inline cli/src/commands/feature-flag/commands/recompose.ts:90]: Is `try/catch` necessary here? There's a top-level function `captureCommandFailure` which handles this I believe.

File: cli/test/feature-flag/utils.ts [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import type { PartialMessage } from '@bufbuild/protobuf';
```
Comment (comatory) [inline cli/test/feature-flag/utils.ts:1]: I don't think we need these utils in separate file.

**Review by JivusAyrus — APPROVED** [review]

**Review by Aenimus — APPROVED** [review]
LGTM!

**Review by comatory — COMMENTED** [review]
Sorry I finished the review just now. I have only some suggestions for improvement.
